src/pinned_memory_manager.h

```diff
@@ -77,6 +77,9 @@ class PinnedMemoryManager {
   // Return Status object indicating success or failure.
   static Status Free(void* ptr);
 
+  static uint64_t GetTotalPinnedMemoryByteSize();
+  static uint64_t GetUsedPinnedMemoryByteSize();
```

Comment:
Building upon my previous suggestion, as this static function can operate on static variable, you can create a static set in this class that holds all the PinnedMemoryManager objects created and individually accumulates the values from the specific objects.

This suggestion ensures that allocate and deallocate are always accompanied by allocated_memory_info update.

Reply:
Additionally in the destructor you'd have to ensure that object is removed from this static set.